src/NuGet.Services.Metadata.Catalog.Monitoring/Resources/PackageRegistrationMetadataResourceV3.cs

```diff
@@ -26,31 +27,52 @@ public class PackageRegistrationMetadataResourceV3 : IPackageRegistrationMetadat
 
         public async Task<PackageRegistrationIndexMetadata> GetIndex(PackageIdentity package, ILogger log, CancellationToken token)
         {
-            var feedPackage = await GetPackageFromIndex(package, log, token);
-            return feedPackage != null ? 
-                JsonConvert.DeserializeObject<PackageRegistrationIndexMetadata>(feedPackage.ToString()) : 
-                null;
+            try
+            {
+                var feedPackage = await GetPackageFromIndex(package, log, token);
+                return feedPackage != null ?
+                    JsonConvert.DeserializeObject<PackageRegistrationIndexMetadata>(feedPackage.ToString()) :
+                    null;
+            }
+            catch (Exception e)
+            {
+                throw new ValidationException($"Could not fetch {nameof(PackageRegistrationIndexMetadata)} from V3!", e);
+            }
         }
 
         public async Task<PackageRegistrationLeafMetadata> GetLeaf(PackageIdentity package, ILogger log, CancellationToken token)
         {
-            var feedPackage = await GetPackageFromLeaf(package, log, token);
-            return feedPackage != null ?
-                JsonConvert.DeserializeObject<PackageRegistrationLeafMetadata>(feedPackage.ToString()) :
-                null;
+            try
+            {
+                var feedPackage = await GetPackageFromLeaf(package, log, token);
+                return feedPackage != null ?
+                    JsonConvert.DeserializeObject<PackageRegistrationLeafMetadata>(feedPackage.ToString()) :
+                    null;
+            }
+            catch (Exception e)
+            {
```

Comment:
The purpose of this try-catch is to tell the person debugging the package that "hey the issue is with this part of the code". Catching a more specific exception type doesn't really add any value because I want it to do the same thing--rethrow the exception with a more specific message--regardless of the actual exception.